uuid_extension.c

```diff
@@ -0,0 +1,224 @@
+        if (isxdigit(zStr[0]) && isxdigit(zStr[1])) {
+            aBlob[i] = (sqlite3UuidHexToInt(zStr[0]) << 4) + sqlite3UuidHexToInt(zStr[1]);
+            zStr += 2;
+        } else {
+            return 1;
+        }
+    }
+    if (zStr[0] == '}')
+        zStr++;
+    return zStr[0] != 0;
+}
+
+/*
+ * Render sqlite3_value pIn as a 16-byte UUID blob.  Return a pointer
+ * to the blob, or NULL if the input is not well-formed.
+ */
+static const unsigned char* sqlite3_uuid_input_to_blob(sqlite3_value* pIn, /* Input text */
+                                                       unsigned char* pBuf /* output buffer */
+) {
+    switch (sqlite3_value_type(pIn)) {
+        case SQLITE_TEXT: {
+            const unsigned char* z = sqlite3_value_text(pIn);
+            if (sqlite3_uuid_str_to_blob(z, pBuf))
+                return 0;
+            return pBuf;
+        }
+        case SQLITE_BLOB: {
+            int n = sqlite3_value_bytes(pIn);
+            return n == 16 ? sqlite3_value_blob(pIn) : 0;
+        }
+        default: {
+            return 0;
+        }
+    }
+}
+
+/*
+ * uuid_generate generates a version 4 UUID as a string
+ */
+static void uuid_generate(sqlite3_context* context, int argc, sqlite3_value** argv) {
+    unsigned char aBlob[16];
+    unsigned char zStr[37];
+    (void)argc;
+    (void)argv;
+    sqlite3_randomness(16, aBlob);
+    aBlob[6] = (aBlob[6] & 0x0f) + 0x40;
+    aBlob[8] = (aBlob[8] & 0x3f) + 0x80;
+    sqlite3_uuid_blob_to_str(aBlob, zStr);
+    sqlite3_result_text(context, (char*)zStr, 36, SQLITE_TRANSIENT);
+}
+
+/*
+ * uuid_str converts a UUID X into a well-formed UUID string.
+ * X can be either a string or a blob.
+ */
+static void uuid_str(sqlite3_context* context, int argc, sqlite3_value** argv) {
+    unsigned char aBlob[16];
+    unsigned char zStr[37];
+    const unsigned char* pBlob;
+    (void)argc;
+    pBlob = sqlite3_uuid_input_to_blob(argv[0], aBlob);
+    if (pBlob == 0)
+        return;
+    sqlite3_uuid_blob_to_str(pBlob, zStr);
+    sqlite3_result_text(context, (char*)zStr, 36, SQLITE_TRANSIENT);
+}
+
+/*
+ * uuid_blob converts a UUID X into a 16-byte blob.
+ * X can be either a string or a blob.
+ */
+static void uuid_blob(sqlite3_context* context, int argc, sqlite3_value** argv) {
+    unsigned char aBlob[16];
+    const unsigned char* pBlob;
+    (void)argc;
+    pBlob = sqlite3_uuid_input_to_blob(argv[0], aBlob);
+    if (pBlob == 0)
+        return;
+    sqlite3_result_blob(context, pBlob, 16, SQLITE_TRANSIENT);
+}
+
+int sqlite_uuid_init(sqlite3* db) {
+    static const int flags = SQLITE_UTF8 | SQLITE_INNOCUOUS;
+    static const int det_flags = SQLITE_UTF8 | SQLITE_INNOCUOUS | SQLITE_DETERMINISTIC;
+    sqlite3_create_function(db, "sqlite_fdw_gen_random_uuid", 0, flags, 0, uuid_generate, 0, 0);
+    sqlite3_create_function(db, "sqlite_fdw_uuid_str", 1, det_flags, 0, uuid_str, 0, 0);
+    sqlite3_create_function(db, "sqlite_fdw_uuid_blob", 1, det_flags, 0, uuid_blob, 0, 0);
```

Comment:
What happened if we cannot create `sqlite_fdw_uuid_blob`. Is there any check?   
This function always returns `SQLITE_OK`.

Reply:
Rewritten to _can not establish connection_ in case of error of creating UUID function. Please ensure my algorithm is ok.

Reply:
Thank for fixing. Confirmed.